File: composeApp/src/androidMain/kotlin/dem/dev/timeflame/feature/main/state/CalendarViewState.kt
package dem.dev.timeflame.feature.main.state

enum class CalendarViewState {
    WEEK, MONTH
}

fun CalendarViewState.toggle(): CalendarViewState {
    return when(this) {
        CalendarViewState.MONTH -> CalendarViewState.WEEK
        CalendarViewState.WEEK -> CalendarViewState.MONTH
    }
}

File: iosApp/iosApp/feature/main/util/CalendarViewState.swift
//
//  CalendarViewState.swift
//  iosApp
//
//  Created by Mikhail Maevskii on 15/12/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation

enum CalendarViewState {
    case Week
    case Month
}

extension CalendarViewState {
    func toggle() -> CalendarViewState {
        return switch self {
        case CalendarViewState.Month:
            CalendarViewState.Week
        case CalendarViewState.Week:
            CalendarViewState.Month
        }
    }
}
